### PR TITLE
Add support for Set/Get map leaves in mysql storage.

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: storage/mysql/map_storage.go [hunk 1]
```diff
@@ -67,12 +74,54 @@ type mapTX struct {
...
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
+
+	_, err = stmt.Exec(m.ms.mapID.TreeID, []byte(keyHash), m.writeRevision, flatValue)
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (Martin2112) [inline storage/mysql/map_storage.go]: Could just return err unless you're planning to log it.

File: storage/mysql/storage_test.go [hunk 2]
```diff
@@ -889,6 +889,145 @@ func TestMapRootUpdate(t *testing.T) {
...
+func TestMapSetSameKeyInSameRevisionFails(t *testing.T) {
+	cleanTestDB()
+
+	mapID := createMapID("TestMapSetSameKeyInSameRevisionFails")
+	db := prepareTestMapDB(mapID, t)
+	defer db.Close()
+	s := prepareTestMapStorage(mapID, t)
+
+	keyHash := []byte("A Key Hash")
+	value := trillian.MapLeaf{
```
Comment (Martin2112) [inline storage/mysql/storage_test.go]: Could pull this value out as it's used in a couple of places.

File: storage/mysql/storage_test.go [hunk 3]
```diff
@@ -889,6 +889,145 @@ func TestMapRootUpdate(t *testing.T) {
...
+
+	readRev := int64(1)
+
+	{
+		tx := beginMapTx(s, t)
+
+		if err := tx.Set(keyHash, value); err != nil {
+			t.Fatalf("Failed to set %v to %v: %v", keyHash, value, err)
+		}
+		tx.Commit()
```
Comment (Martin2112) [inline storage/mysql/storage_test.go]: I thought commit() could fail?

**Comment by Martin2112** [comment]
Looks basically OK, couple of suggestions.

File: storage/mysql/map_storage.go (hunk 1 above)
Comment (AlCutter) [inline storage/mysql/map_storage.go]: done

File: storage/mysql/storage_test.go (hunk 2 above)
Comment (AlCutter) [inline storage/mysql/storage_test.go]: Done

File: storage/mysql/storage_test.go (hunk 3 above)
Comment (AlCutter) [inline storage/mysql/storage_test.go]: Add checks for these new ones, but there are a *lot* of unchecked Commit() called in here, I've dropped a `TODO` in there to go back and sort those out too.

**Pull request merged**